MCP server fails to start in JetBrains IDEs — bare `jbang` and `java` commands not found (ENOENT)

### Problem

The `quarkus-agent-mcp` server fails to start when configured in JetBrains IDEs (IntelliJ IDEA) with the GitHub Copilot extension. The same configuration works correctly in VS Code.

### Error

```
Error: spawn jbang ENOENT
Failed to connect to MCP server: spawn jbang ENOENT
```

The same `ENOENT` error occurs when using `java -jar` instead of `jbang`:

```json
{
  "quarkus-agent": {
    "type": "stdio",
    "command": "java",
    "args": ["-jar", "/path/to/quarkus-agent-mcp-runner.jar"]
  }
}
```

### Root Cause

JetBrains IDEs launched from the desktop environment inherit the **system PATH** (from `systemd`), not the **shell PATH** (from `~/.bashrc` / `~/.zshrc` / `~/.profile`). Binaries installed in user-local locations such as `~/.jbang/bin/` or `~/.sdkman/candidates/java/current/bin/` are on the shell PATH but not on the system PATH — so the IDE process cannot find them.

This is different from VS Code, which typically inherits the full shell environment.

This creates a confusing situation where IntelliJ's **built-in terminal** finds `jbang`/`java` perfectly (because it starts a proper shell that sources `~/.bashrc`), but the MCP server spawned by the IDE fails because it only sees the system PATH.

As Max Andersen noted: *"You need jbang in the system path for it to work — same with any other tool called from IntelliJ."*

The same applies to `java` when installed via user-local version managers (SDKMAN!, asdf, mise, etc.).

### Reproduction

1. Install JBang via the default method (installs to `~/.jbang/bin/` — shell PATH, not system PATH)
2. Configure `quarkus-agent-mcp` in JetBrains Copilot MCP settings:

```json
{
  "quarkus-agent": {
    "type": "stdio",
    "command": "jbang",
    "args": ["quarkus-agent-mcp@quarkusio"]
  }
}
```

3. Restart the IDE — the MCP server fails with `spawn jbang ENOENT`

### Suggested Improvement

**Update the README** with a JetBrains-specific installation section that explains the shell PATH vs. system PATH distinction, and that `jbang` (or `java`) must be on the **system PATH** for the MCP server to work in JetBrains IDEs.

### Environment

- IDE: IntelliJ IDEA (Community / Ultimate) with GitHub Copilot extension
- OS: Arch Linux with Hyprland (Wayland compositor)
- JBang: installed at `~/.jbang/bin/jbang` (shell PATH, not system PATH)
- Java: also affected when installed via user-local version managers (SDKMAN!, asdf, mise, etc.)
- Works correctly in: VS Code

I did some debugging and testing.

```
{
  "servers": {
    "quarkus-agent2": {
      "type": "stdio",
      "command": "java",
      "args": ["-jar", "C:\\ieu\\quarkus-agent-mcp-1.0.1-SNAPSHOT-runner.jar"]
    }
  }
}
```
Works, when I start the IDE from the console (Powershell).

Even with Java in the System Path, it does not work otherwise, but exits with the following error
```
[2026-04-21 14:02:37.742][info]    [quarkus-agent2]Starting server quarkus-agent2
[2026-04-21 14:02:38.356][warning] [quarkus-agent2][server stderr] LogManager error of type GENERIC_FAILURE: Section header cannot be encoded into charset "windows-1252"
[2026-04-21 14:02:38.357][warning] [quarkus-agent2][server stderr] 
[2026-04-21 14:02:38.407][warning] [quarkus-agent2][server stderr] 2026-04-21 14:02:38,394 ERROR [io.quarkus.runtime.Application] (main) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:112)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:127)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:50)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:143)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.nio.file.AccessDeniedException: C:\WINDOWS\system32\config
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:89)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
	at java.base/sun.nio.fs.WindowsDirectoryStream.<init>(WindowsDirectoryStream.java:86)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.newDirectoryStream(WindowsFileSystemProvider.java:541)
	at java.base/java.nio.file.Files.newDirectoryStream(Files.java:614)
	at io.quarkus.runtime.configuration.ConfigDiagnostic.configFiles(ConfigDiagnostic.java:224)
	at io.quarkus.runtime.configuration.ConfigDiagnostic.configFilesFromLocations(ConfigDiagnostic.java:239)
	at io.quarkus.runtime.configuration.ConfigRecorder.unknownConfigFiles(ConfigRecorder.java:107)
	at io.quarkus.runner.recorded.ConfigGenerationBuildStep$unknownConfigFiles604069353.deploy_0(Unknown Source)
	at io.quarkus.runner.recorded.ConfigGenerationBuildStep$unknownConfigFiles604069353.deploy(Unknown Source)
	... 7 more
[2026-04-21 14:02:38.448][info]    [quarkus-agent2]Connection state: Stopped
[2026-04-21 14:02:38.450][error]   [quarkus-agent2]Failed to connect to MCP server: MCP error -32000: Connection closed
[2026-04-21 14:02:38.450][error]   [quarkus-agent2]message: MCP error -32000: Connection closed stack: McpError: MCP error -32000: Connection closed
    at Function.fromError (C:\snapshot\copilot-intellij\node_modules\@modelcontextprotocol\sdk\src\types.ts:2316:16)
    at Sct._onclose (C:\snapshot\copilot-intellij\node_modules\@modelcontextprotocol\sdk\src\shared\protocol.ts:657:32)
    at Fct._transport.onclose (C:\snapshot\copilot-intellij\node_modules\@modelcontextprotocol\sdk\src\shared\protocol.ts:618:18)
    at ChildProcess.<anonymous> (C:\snapshot\copilot-intellij\node_modules\@modelcontextprotocol\sdk\src\client\stdio.ts:143:22)
    at ChildProcess.emit (node:events:519:28)
    at ChildProcess.t.emit (C:\snapshot\copilot-intellij\node_modules\cross-spawn\lib\enoent.js:34:29)
    at maybeClose (node:internal/child_process:1101:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
```

IMO this is more a bug/missing feature at Intellij/Copilot Plugin than the quarkus-agent-mcp. I'm not even sure if we would be able to work around it.

--- a/README.md
+++ b/README.md
@@ -98,6 +98,38 @@ Add to `~/.codeium/windsurf/mcp_config.json`:
 }
 ```
 
+#### JetBrains IDEs (IntelliJ IDEA, WebStorm, etc.)
+
+In **Settings → Tools → AI Assistant → MCP Servers** (or the GitHub Copilot MCP settings), add a server with an **absolute path** to `jbang`:
+
+```json
+{
+  "servers": {
+    "quarkus-agent": {
+      "type": "stdio",
+      "command": "/home/you/.jbang/bin/jbang",
+      "args": ["quarkus-agent-mcp@quarkusio"]
+    }
+  }
+}
+```
+
+> **Why absolute paths?** JetBrains IDEs launched from the desktop inherit the **system PATH**, not your shell PATH (`~/.bashrc` / `~/.zshrc`). Tools installed via SDKMAN!, JBang's default installer, asdf, or mise are typically on the shell PATH only — so the IDE can't find them. See [Troubleshooting](#jbang-or-java-not-found-enoent) below.
+
+Alternatively, use the direct download method with an absolute path to `java`:
+
+```json
+{
+  "servers": {
+    "quarkus-agent": {
+      "type": "stdio",
+      "command": "/path/to/java",
+      "args": ["-jar", "/path/to/quarkus-agent-mcp-runner.jar"]
+    }
+  }
+}
+```
+
 ### Via direct download
 
 Download the uber-jar from the [latest GitHub Release](https://github.com/quarkusio/quarkus-agent-mcp/releases/latest), then:
@@ -339,6 +371,63 @@ open presentation/index.html        # macOS
 ```
 
 Press **F** for fullscreen, arrow keys or space to navigate slides.
+
+## Troubleshooting
+
+### `jbang` or `java` not found (ENOENT)
+
+```
+Error: spawn jbang ENOENT
+```
+
+This happens when the MCP client (IDE, desktop app) cannot find `jbang` or `java` on its PATH. It affects **JetBrains IDEs**, **Claude Desktop**, and other desktop applications that don't inherit your shell environment.
+
+**Why it happens:** Desktop applications launched from a graphical environment inherit the **system PATH** (from `systemd` on Linux, or system environment variables on Windows/macOS), not the **shell PATH** (from `~/.bashrc`, `~/.zshrc`, `~/.profile`). Tools installed via user-local version managers — SDKMAN!, JBang's default installer, asdf, mise — are on the shell PATH but not the system PATH.
+
+This is different from VS Code, which typically resolves the full shell environment, and from IDE built-in terminals, which start a proper login shell.
+
+**Solutions (pick one):**
+
+1. **Use absolute paths** in your MCP config (recommended):
+   ```json
+   {
+     "command": "/home/you/.jbang/bin/jbang",
+     "args": ["quarkus-agent-mcp@quarkusio"]
+   }
+   ```
+   Find the path with `which jbang` or `which java` in a terminal.
+
+2. **Symlink into a system PATH directory:**
+   ```bash
+   sudo ln -s ~/.jbang/bin/jbang /usr/local/bin/jbang
+   ```
+
+3. **Add to the system PATH permanently:**
+   - **Linux (systemd):** Create `/etc/profile.d/jbang.sh` with `export PATH="$HOME/.jbang/bin:$PATH"`, or add the path to `~/.config/environment.d/` for systemd user sessions.
+   - **macOS:** Use `launchctl setenv PATH ...` or add to `/etc/paths.d/`.
+   - **Windows:** Add `%USERPROFILE%\.jbang\bin` to the system environment variables (System → Advanced → Environment Variables → System variables → Path).
+
+### `AccessDeniedException: C:\WINDOWS\system32\config` (Windows)
+
+```
+java.nio.file.AccessDeniedException: C:\WINDOWS\system32\config
+```
+
+This is a [Quarkus core issue](https://github.com/quarkusio/quarkus/issues/53739) that occurs on Windows when the MCP server is launched by an IDE with the working directory set to `system32`. Quarkus scans `{working-dir}/config` for configuration files and hits the protected Windows registry hive directory.
+
+**Workaround:** Set the working directory explicitly in your MCP config, or pass `-Duser.dir` to override it:
+
+```json
+{
+  "servers": {
+    "quarkus-agent": {
+      "type": "stdio",
+      "command": "java",
+      "args": ["-Duser.dir=C:\\Users\\you", "-jar", "C:\\path\\to\\quarkus-agent-mcp-runner.jar"]
+    }
+  }
+}
+```
 
 ## Related Projects
 

